interbotix_common_toolbox/interbotix_gravity_compensation/src/interbotix_gravity_compensation_obj.cpp

```diff
@@ -0,0 +1,475 @@
+    for (size_t i = 0; i < joint_names_.size(); i++) {
+      torque_enable("single", joint_names_[i], true);
+      torque_enable_requests_[i] = false;
+    }
+  }
+
+  // Set the gravity compensation flag and return the response
+  response->success = true;
+  response->message = std::string("Gravity Compensation ");
+  response->message += request->data ? "Enabled" : "Disabled";
+  RCLCPP_INFO(this->get_logger(), response->message.c_str());
+}
+
+bool InterbotixGravityCompensation::load_motor_specs(const std::string & motor_specs)
+{
+  // Load the torque constants and current units from the motor_specs file
+  YAML::Node motor_specs_node;
+  try {
+    motor_specs_node = YAML::LoadFile(motor_specs);
+  } catch (YAML::Exception & e) {
+    RCLCPP_ERROR(
+      this->get_logger(),
+      "Failed to load the motor specs file '%s'! Error: %s",
+      motor_specs.c_str(), e.what());
+    return false;
+  }
+
+  // Load the joint names and joint name to index mapping
+  for (size_t i = 0; i < motor_specs_node["joint_names"].size(); i++) {
+    std::string joint_name = motor_specs_node["joint_names"][i].as<std::string>();
+    joint_names_.push_back(joint_name);
+    joint_name_to_index_[joint_name] = i;
+  }
+
+  // Load the torque constants, current units, and no load currents
+  float all = motor_specs_node["motor_assist"]["all"].as<float>();
+  float single = 0.0;
+  for (const auto & joint_name : joint_names_) {
+    if (motor_specs_node["motor_assist"][joint_name].IsDefined()) {
+      // Load the motor specs
+      torque_constants_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["torque_constant"].as<float>()
+      );
+      current_units_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["current_unit"].as<float>()
+      );
+      no_load_currents_.push_back(
+        motor_specs_node["motor_specs"][joint_name]["no_load_current"].as<float>()
+      );
+
+      // Scale the no load current according to the motor assist setting
+      if (all == -1) {
+        single = motor_specs_node["motor_assist"][joint_name].as<float>();
+        if (0 <= single && single <= 1) {
+          no_load_currents_.back() *= single;
+        } else {
+          RCLCPP_WARN(
+            this->get_logger(),
+            "Motor assist value not in the range [0, 1] for joint %s. Setting it to 0.",
+            joint_name.c_str());
+          no_load_currents_.back() *= 0;
+        }
+      } else if (0 <= all && all <= 1) {
+        no_load_currents_.back() *= motor_specs_node["motor_assist"]["all"].as<float>();
+      } else {
+        RCLCPP_WARN(
+          this->get_logger(),
+          "Motor assist value not in the range [0, 1] or -1 for all joints. Setting it to 0.");
+        no_load_currents_.back() *= 0;
+      }
+    } else {
+      RCLCPP_WARN(
+        this->get_logger(),
+        "Motor specs not found for joint %s in '%s', "
+        "because it cannot run in the Current Control Mode. "
```

Comment:
State that we are assuming that the joint does not support current control mode.